### ISSUE TITLE
Google Interview University

### ISSUE THREAD

**Issue by nelsonic** [dwyl/start-here#99]
While I may not _understand/agree_ with the _motives_ of this quest (_Google is not the company it once was... just read the posts on [Quora](https://www.quora.com/topic/Working-at-Google-1) ..._) I totally _admire_ the dedication/effort @jwasham is pouring into his mission: https://github.com/jwasham/google-interview-university & https://googleyasheck.com/

I cannot help but think that the _Goal_ of working for a specific company rather on solving a specific _problem_ is putting the cart before the horse. and _instead_ he should select the _problem_ and then find an organisation that is tackling it ... but, to be fair, if there is _any_ problem _worth solving_ Alphabet has _probably_ already got a team tackling it ... so maybe getting the skills first is the way to go...? 💭

**Comment by jwasham** [comment]
Hi @nelsonic,
This issue popped up for me because (I think) my name was mentioned. So I appeared!

I've worked for many companies and even started a few of my own. There's no perfect company because there are no perfect people. I wanted to make a career change and level up my skills, not just for Google, but for any company I choose to work for. It's about choice, and Google is the most ambitious target I could find.

I've read much about the company and talked to employees. I respect Google's culture, and from it many of the stances they have taken, and their requirement for professionalism and excellence in building software.

Even if I don't get the job, I will have gained sufficient practical knowledge and skills that I can apply to any problem I choose to solve, wherever I wish.

**Comment by nelsonic** [comment]
@jwasham, yeah, the skills you are _relentlessly_ acquiring are pretty much universally transferrable. 
Given how much publicity you've given them with you superb learning resource they would _mad_ not to hire you and split your time between engineering and "_evangelism_" (_I mean that in the best possible way_). 
I've read _everything_ I can about Google (_books, articles, Q&A_) and really admire them! 
when a GTech Recruiter approached me in 2011 I was _seriously_ tempted to go work for them ...
(_I kept the email exchange filed away and hope I never regret not making the move...!_)

The people who work for google are _mostly_ amazing (_smart, yet humble and understated_)
but the _culture_ has shifted away from "scrappy startup" to "enterprise" which means management, meetings and performance reviews. (_I cannot bare the idea of being [micro]managed by a mid-level manager who last wrote code in the 90's and only cares about a "burn down chart" and wether you filed your "TPS Report"..._ [M&Ms](https://www.ted.com/talks/jason_fried_why_work_doesn_t_happen_at_work) ) 😢 

If given the choice between Google and another "Corporate" the choices is _obvious_, but too often people _limit_ themselves too narrowly when their actual options are _way_ more exciting!

> "_Do you want to spend your time getting people to click on ads or do you want to change the world?_"

Of all the things you've learned over the last few months, what were you most _fascinated_ by and which technology or tool sparked the most _creativity_? _surely_ you've thought of an _idea_ for a project you can work on _without_ having to work for a "Big Company" ...? 😉 

Either way, hats off John! you have amassed more CS knowledge than most CS grads! 
When I retire next year I will go through a few of the items on your list. (_thanks for sharing it!_)

**Comment by jwasham** [comment]
Of all the many interesting items I've studied, one of the fascinating things was machine learning.  Not required for a Google interview, but they are greatly increasing their use of that area, as you can imagine. It can be super mathy, but it's manageable.

I've started four companies, two are still going. Starting and growing a company is terribly difficult. I'm not looking to start any more.

I'm excited to work on big projects, far outside my experience and comfort zone.

Thanks again, Nelson!
